sharedbuffers/mapped_struct.py

```diff
@@ -6517,48 +6540,43 @@ def build(cls, initializer, destfile = None, tempdir = None,
             bigparts = []
             parts = []
             islice = itertools.islice
-            array = numpy.array
             curpos = basepos + cls._Header.size
-            part = []
             struct_dt = numpy.dtype([
                 ('key', dtype),
                 ('value', dtype),
             ])
+            ppartbuf = partbuf = numpy.empty((1000000, 2), npuint64)
```

Comment:
Why did you choose this buffer size? Can't we use a constant for the buffer size?

Reply:
It's about the size at which malloc starts using mmap to allocate stuff, which means deallocation actually frees memory immediately to the OS (munmap). Smaller allocations go to the heap and get put on a freelist instead when deallocated, which does not return memory to the OS.

The actual threshold is dynamic but my experience is that it tends to roam the 8MB range.

I'm not sure hypertuning this would help much, the impact isn't huge and the optimal size would depend a lot on the use case.

A constant... well, why not, but I don't have a good name for it. :thinking: